src/libxtp/calculators/ianalyze.cc

```diff
@@ -22,48 +22,42 @@
 #include <votca/tools/histogramnew.h>
 #include <votca/xtp/topology.h>
 
-using namespace std;
-
 namespace votca {
 namespace xtp {
 
-void IAnalyze::Initialize(tools::Property &opt) {
-  std::cout << std::endl;
-  _do_pairtype = false;
-  _do_IRdependence = false;
-  std::string key = "options." + Identify();
-  if (opt.exists(key + ".states")) {
-    std::string statestrings = opt.get(key + ".states").as<std::string>();
-    tools::Tokenizer tok(statestrings, ",\n\t ");
-    std::vector<std::string> string_vec;
-    tok.ToVector(string_vec);
-    for (std::string &state : string_vec) {
-      _states.push_back(QMStateType(state));
-    }
-  } else {
-    _states.push_back(QMStateType(QMStateType::Electron));
-    _states.push_back(QMStateType(QMStateType::Hole));
+void IAnalyze::Initialize(tools::Property &user_options) {
+
+  // get pre-defined default options from VOTCASHARE/xtp/xml/eanalyze.xml
+  LoadDefaults("xtp");
+  // update options with user specified input
+  UpdateWithUserOptions(user_options);
```

Comment:
this should be at a lower level, because it is not special for ianalyze

Reply:
What do you mean by lower level :) ? like in a parent class method?

Reply:
yes exactly. See below.